### PR TITLE
move mixed str/series groupby case handling to a backend

### PR DESCRIPTION
**Pull request by ienkovich**
It looks like pandas-in-ray cannot process `by` frames we create on API level for `groupby` calls when a mix of column names and `Series` objects is used in `by`. Initially, we did it to avoid `default_to_pandas` call in `DataFrameGroupBy`. I moved this call into `GroupbyReduceFunction`. This also let me modify `DataFrameGroupBy.size` and remove `default_to_pandas` call from there. This allowed me to roll back our changes from `DataFrame.groupby` and handle this tricky `by` case in our engine.

### CODE REVIEW

**Comment by Garra1980** [comment]
Do we have all necessary coverage by tests already?

**Comment by ienkovich** [comment]
Since there is no new functionality or a bug fix I don't add any tests. Checked both backends work on NYC Taxi Q3 though.

File: modin/pandas/dataframe.py [hunk 1]
```diff
@@ -479,22 +479,11 @@ def groupby(
...
-                    cols = self._query_compiler.getitem_column_array(cols)
-                    if cols is None:
-                        raise NotImplementedError("Unsupported groupby arguments")
-                    by = cols.concat(1, series, ignore_index=True)
-
+                elif mismatch and any(
+                    isinstance(obj, str) and obj not in self.columns for obj in by
+                ):
+                    names = [o.name if isinstance(o, Series) else o for o in by]
+                    raise KeyError(next(x for x in names if x not in self))
```
Comment (Garra1980) [inline modin/pandas/dataframe.py:486]: I'm wondering what do we do in "else" case here...

Comment (ienkovich) [reply modin/pandas/dataframe.py:486]: Here we just look for invalid column names and raise an error. This patch just reverts our previous changes in this function and it now matches the master branch.

**Comment by Garra1980** [comment]
> Since there is no new functionality or a bug fix I don't add any tests. Checked both backends work on NYC Taxi Q3 though.
I understand that but we changed some conditions which could change coverage. Anyway these are mostly changes in general part so I suppose Modin maintainers should decide ;)

**Review by Garra1980 — APPROVED** [review]

**Pull request merged**